recipes/pydevtool/meta.yaml

```
@@ -0,0 +1,44 @@
+{% set name = "pydevtool" %}
+{% set version = "0.2.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/pydevtool-{{ version }}.tar.gz
+  sha256: e821ce0b637b1bbf7638b1de23522b76c23b8e9900678ff46e6c846dcd9464e2
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python >=3.8
+  run:
+    - doit
+    - python >=3.8
+
+test:
+  imports:
+    - pydevtool
+  commands:
+    - pip check
```

I don't know pydevtool, but from looking at the source, it mainly (or even predominantly) is made up of a CLI.

I'd suggest that some common commands be added to he `commands:` section.
```suggestion
  commands:
    - pip check
    - pydevtool something something
```

---

It is not a CLI itself (for now), but provide some helpers to build CLIs.

---

I disagree with this point. It's not the responsibility of a package manager to test the package but the packaging. Plus it's a simple Python set of functions. Hence if we can import the namespace, from a packaging point of view, the job is done. If there are other issues there are to be fixed/found on the package repo.

---

Well... If only the packaging were independent of the functionality of the package.

As it is, it is best practice to run the upstream test suite for the feedstock, in order to make sure no functionality has been inadvertently broken by the packaging (and you can believe me from many many many painful experiences, that a package import is not even close to an adequate proxy for correct functionality - more like the barest minimum).

That said, there are cases where there's indeed not much to test (and this may be one of them), but I disagree in turn with your characterization of the responsibilities of the package manager (which isn't conda-specific; distros and other package managers generally also use the project's test suites to likewise ensure functionality of their packaging).

---

> Well... If only the packaging were independent of the functionality of the package.

We are talking about a pure Python package which consist in a single file. I stand by my statement.

---

> ... other package managers generally also use the project's test suites to likewise ensure functionality of their packaging).

Exactly my point. It's not the responsibility of individual packages to do that. Package managers can use these package if they want in their own test suite.

---

Anyway, feel free to add a suggestion. I don't plan to do more now.

---

As you will find out after enough time in packaging, that is not a sustainable approach long-term. Packages evolve, the packaging often doesn't - until something breaks somewhere. The work to add some proper testing is a good investment compared to fixing the fallout of a breakage later (even when factoring in ~low likelihood), but well; some things need to be experienced I guess.

---

[some of your comments arrived in my browser after my last comment]

> > ... other package managers generally also use the project's test suites to likewise ensure functionality of their packaging).
> 
> Exactly my point. It's not the responsibility of individual packages to do that. Package managers can use these package if they want in their own test suite.

conda/mamba + conda-forge is a package manager, or maybe we have a crossed wire somewhere? People developing using conda-forge packages (like scipy does) expect correct functionality - where should that be ensured IYO? With our current infrastructure, there's nothing _but_ the feedstocks (for individual packages) that can reasonably ensure correctness of each consumable output.

> Anyway, feel free to add a suggestion. I don't plan to do more now.

Fine by me